SwiftLeeds/Views/Components/FancyHeaderView.swift

```diff
@@ -0,0 +1,109 @@
+//
+//  HeaderView.swift
+//  SwiftLeeds
+//
+//  Created by Kannan Prasad on 05/07/2022.
+//
+
+import SwiftUI
+
+struct FancyHeaderView: View {
+    let title: String
+    let foregroundImageURL: URL
+    let backgroundImageURL: URL
+    
+    private let foregroundImageWidth = 160.0
+    private let aspectRatio = 1.66
+    
+    @State var foregroudGroupViewHeight: CGFloat = .zero
+    
+    var body: some View {
+        Rectangle()
+            .foregroundColor(.clear)
+            .edgesIgnoringSafeArea(.top)
+            .aspectRatio(aspectRatio, contentMode: .fill)
+            .background(backgroundImage
+                            .aspectRatio(contentMode: .fill))
+            .overlay(foregroudGroupView,alignment: .center)
+            .padding(.bottom,foregroudGroupViewHeight/2)
+    }
+    
+    private var backgroundImage: some View {
+        AsyncImage(url: backgroundImageURL) { phase in
+            
+            switch phase {
+            case .empty:
+                ProgressView()
+            case .success(let image):
+                image
+                    .resizable()
+                    .aspectRatio(aspectRatio, contentMode: .fill)
+            case .failure(_):
+                ProgressView()
```

Comment:
The outer AsyncImage doesn't have `aspectRatio` or any sizing set on it, so in this instance the sizing will be changing as the content renders. I generally find its better to have a Rectangle with a fixed size, and then put the `AsyncImage` in its background. This way, you don't have sizes jumping around, and you're safe to do what you want inside this block.

Reply:
Created a method to return image to embed with in a rectangle as background.